src/analysis/multiSpecies/microbiomeModelingToolbox/mgPipe/initMgPipe.m

```diff
@@ -30,58 +30,119 @@
 %    abunFilePath:           char with path and name of file from which to retrieve abundance information
 %    objre:                  char with reaction name of objective function of organisms
 %    figForm:                format to use for saving figures
-%    numWorkers:             logical indicating the number of cores to use for parallelization
+%    numWorkers:             boolean indicating the number of cores to use for parallelization
 %    autoFix:                double indicating if to try to automatically fix inconsistencies
-%    compMod:                logical indicating if outputs in open format should be produced for each section (1=T)
-%    patStat:                logical indicating if documentation on health status is available
-%    rDiet:                  logical indicating if to enable also rich diet simulations
-%    extSolve:               logical indicating if to save the constrained models to solve them externally
-%    fvaType:                logical indicating which function to use for flux variability
-%    autorun:                logical used to enable /disable autorun behavior (please set to 1)
+%    compMod:                boolean indicating if outputs in open format should be produced for each section (1=T)
+%    patStat:                boolean indicating if documentation on health status is available
+%    rDiet:                  boolean indicating if to enable also rich diet simulations
+%    extSolve:               boolean indicating if to save the constrained models to solve them externally
+%    fvaType:                boolean indicating which function to use for flux variability
+%    autorun:                boolean used to enable /disable autorun behavior (please set to 1)
 %
 % .. Author: Federico Baldini 2018
 
-init = 0;
+global CBTDIR
+
+init = false;
+
+% path checking
+if ~exist('modPath', 'var') || ~exist(modPath, 'dir')
```

Comment:
If these are all optional inputs now, they should probably be marked as such. 
And I would, due to the mass of options, suggest to use either a struct input, or an inputParser (with the latter allowing both structs and parameters to be used..... )

Reply:
Ok, I will change that in the header. An input parser is for later.